DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,155 @@
+# Developer Guide
```

Comment:
I'm not sure this should be a new file; isn't it targeted at the same audience that the existing CONTRIBUTING.md file is?

Reply:
The CONTRIBUTING.md file seemed for focused on policies rather than getting things running (apart from the "tests" section), but I'm happy to move this content there.